dotnet/src/Plugins/Plugins.Core/ConversationSummaryPlugin.cs

```diff
@@ -59,50 +58,43 @@ public ConversationSummaryPlugin()
     [KernelFunction, Description("Given a long conversation transcript, summarize the conversation.")]
     public Task<string> SummarizeConversationAsync(
         [Description("A long conversation transcript.")] string input,
-        Kernel kernel,
-        ContextVariables variables) =>
-        ProcessAsync(this._summarizeConversationFunction, input, kernel, variables);
+        Kernel kernel) =>
+        ProcessAsync(this._summarizeConversationFunction, input, kernel);
 
     /// <summary>
     /// Given a long conversation transcript, identify action items.
     /// </summary>
     /// <param name="input">A long conversation transcript.</param>
     /// <param name="kernel">The kernel.</param>
-    /// <param name="variables">The context variables for function execution.</param>
     [KernelFunction, Description("Given a long conversation transcript, identify action items.")]
     public Task<string> GetConversationActionItemsAsync(
         [Description("A long conversation transcript.")] string input,
-        Kernel kernel,
-        ContextVariables variables) =>
-        ProcessAsync(this._conversationActionItemsFunction, input, kernel, variables);
+        Kernel kernel) =>
+        ProcessAsync(this._conversationActionItemsFunction, input, kernel);
 
     /// <summary>
     /// Given a long conversation transcript, identify topics.
     /// </summary>
     /// <param name="input">A long conversation transcript.</param>
     /// <param name="kernel">The kernel.</param>
-    /// <param name="variables">The context variables for function execution.</param>
     [KernelFunction, Description("Given a long conversation transcript, identify topics worth remembering.")]
     public Task<string> GetConversationTopicsAsync(
         [Description("A long conversation transcript.")] string input,
-        Kernel kernel,
-        ContextVariables variables) =>
-        ProcessAsync(this._conversationTopicsFunction, input, kernel, variables);
+        Kernel kernel) =>
+        ProcessAsync(this._conversationTopicsFunction, input, kernel);
 
-    private static async Task<string> ProcessAsync(KernelFunction func, string input, Kernel kernel, ContextVariables variables)
+    private static async Task<string> ProcessAsync(KernelFunction func, string input, Kernel kernel)
     {
         List<string> lines = TextChunker.SplitPlainTextLines(input, MaxTokens);
         List<string> paragraphs = TextChunker.SplitPlainTextParagraphs(lines, MaxTokens);
 
         string[] results = new string[paragraphs.Count];
         for (int i = 0; i < results.Length; i++)
         {
-            variables.Update(paragraphs[i]);
-            results[i] = (await func.InvokeAsync(kernel, variables).ConfigureAwait(false)).GetValue<string>() ?? "";
+            results[i] = (await func.InvokeAsync(kernel, new KernelFunctionArguments { { KernelFunctionArguments.InputParameterName, paragraphs[i] } }).ConfigureAwait(false)).GetValue<string>() ?? "";
```

Comment:
Construct the dictionary once and overwrite the value on each iteration?

Reply:
Yep